src/packagedcode/models.py

```diff
@@ -135,153 +210,121 @@ class Package(object):
-        # a URL in the SPDX form of:
-        # git+https://github.com/nexb/scancode-toolkit.git
-        self.vcs_repository = None
-        # a revision, branch, tag reference, etc (can also be included in the URL
-        self.vcs_revision = None
-
-        # a top level copyright often asserted in metadata
-        self.copyright_top_level = None
-        # effective copyrights as detected and eventually summarized
-        self.copyrights = []
-
-        # as asserted licensing information
-        # a list of AssertLicense objects
-        self.asserted_licenses = []
-
-        # list of legal files locations (such as COPYING, NOTICE, LICENSE, README, etc.)
-        self.legal_file_locations = []
-
-        # resolved or detected license expressions
-        self.license_expression = None
-        # a list of license texts
-        self.license_texts = []
-
-        # a list of notices texts
-        self.notice_texts = []
-
-        # map of dependency group to a list of dependencies for each DEPENDENCY_GROUPS
-        self.dependencies = OrderedDict()
-
-        # map to a list of related packages keyed by PAYLOAD
-        # for instance the SRPM of an RPM
-        self.related_packages = OrderedDict()
-
-        # accept all keywords arguments, print a message for unknown arguments
-        for k, v in kwargs.items():
-            # handle properties
-            prop = getattr(self.__class__, k, None)
-            if isinstance(prop, property):
-                prop.fset(self, v)
-                continue
-
-            # plain attributes
-            if not hasattr(self, k):
-                # FIXME: this should be a log message instead
-                print('Warning: creating Package with unknown argument: %(k)r: %(v)r' % locals())
-            setattr(self, k, v)
+    PAYLOADS = [payload_src, payload_bin, payload_doc]
+    # the type of payload in this package. one of PAYLOADS or none
+    payload_type = StringType(choices=PAYLOADS)
+
+    # list of Parties: authors, packager, maintainers, contributors, distributor, vendor, etc
+    authors = ListType(ModelType(Party), default=[])
+    maintainers = ListType(ModelType(Party), default=[])
+    contributors = ListType(ModelType(Party), default=[])
+    owners = ListType(ModelType(Party), default=[])
+    packagers = ListType(ModelType(Party), default=[])
+    distributors = ListType(ModelType(Party), default=[])
+    vendors = ListType(ModelType(Party), default=[])
+
+    # keywords or tags
+    keywords = ListType(StringType(), default=[])
+    # url to a reference documentation for keywords or tags (such as a Pypi or SF.net Trove map)
+    keywords_doc_url = URLType()
+
+    # paths to metadata files for this package, if any
+    # can be the same as the package location (e.g. RPMs)
+    metafile_locations = ListType(StringType(), default=[])
+
+    # URLs to metadata files for this package.
+    metafile_urls = ListType(URLType(), default=[])
+
+    homepage_url = URLType()
```

Comment:
What are the constraints of a URL type? in some case we have a home that is an FTP url. Does it support this? If not then we should use a plain string instead